openpgp-dsm/src/lib.rs

```diff
@@ -818,6 +819,116 @@ pub fn generate_key(
     Ok(())
 }
 
+/// Gets info on a key and prints revelant PGP details for it.
+/// Returns `Err` if key is not present.
+pub fn dsm_key_info(cred: Credentials, key_name: &str) -> Result<Vec<String>> {
+    info!("dsm list_keys");
+    let dsm_client = cred.dsm_client()?;
+    let mut key_info_store: Vec<String> = Vec::new();
+
+    let params = ListSobjectsParams {
+        name: Some(key_name.to_string()),
+        ..Default::default()
+    };
+
+    let keys = dsm_client.list_sobjects(Some(&params))?;
+
+    // If multiple keys with same name exists, iterate and print them all
+    for key in keys.iter() {
+        key_info_store.push(sobject_long_details_formatter(&key));
+    }
+
+    let footer = format!("\nTOTAL OBJECTS: {}\n", key_info_store.len());
+    key_info_store.push(footer);
+
+    Ok(key_info_store)
+}
+/// Iterates through accessible groups and fetches all keys available to app.
+/// Returns an sorted list of keys grouped by group ID.
+/// Note that this funtion has linear time complexity over the number of groups
```

Comment:
Nit: I would drop the last sentence, as this function could be threaded in the future.